apps/build.groovy

```diff
@@ -28,10 +28,48 @@ class AppBuildResults implements Serializable {
 def build(boolean verboseMaven) {
 	dir ('apps') {
 		quietFlags = verboseMaven ? '' : '--quiet --batch-mode'
+  
+  	sh "ACCOUNT_ID=\"$(python3 -c 'import json,sys;print(json.loads(sys.argv[1])[\"Account\"])' \"$(aws sts get-caller-identity)\")\"
+            CODEARTIFACT_ENDPOINT=\"$(aws codeartifact get-repository-endpoint --domain bfd-mgmt --repository bfd-mgmt --format maven --output text)\"
+            CODEARTIFACT_AUTH_TOKEN=\"$(aws codeartifact get-authorization-token --domain bfd-mgmt \
+            --domain-owner \"$ACCOUNT_ID\" \
+            --output text --query authorizationToken)\"
+            cat <<EOF > ~/.m2/settings.xml
+            <settings xmlns=\"http://maven.apache.org/SETTINGS/1.0.0\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"
+            xsi:schemaLocation=\"http://maven.apache.org/SETTINGS/1.0.0 https://maven.apache.org/xsd/settings-1.0.0.xsd\">
+                <profiles>
+                    <profile>
+                        <id>bfd-mgmt-bfd-mgmt</id>
+                        <activation>
+                        <activeByDefault>true</activeByDefault>
+                        </activation>
+                        <repositories>
+                            <repository>
+                                <id>bfd-mgmt-bfd-mgmt</id>
+                                <url>${CODEARTIFACT_ENDPOINT}</url>
+                            </repository>
+                        </repositories>
+                    </profile>
+                </profiles>
+                <servers>
+                    <server>
+                    <username>aws</username>
+                    <password>${CODEARTIFACT_AUTH_TOKEN}</password>
+                    <id>bfd-mgmt-bfd-mgmt</id>
+                    </server>
+                </servers>
+            </settings>
+          EOF"
+	
 
 		sh "mvn ${quietFlags} --threads 1C --update-snapshots -DskipITs -DskipTests -Dmaven.javadoc.skip=true clean verify"
 	}
```

Comment:
I think the following is more in-line with what we've done so far. What do you think?
```suggestion
	withCredentials([string(credentialsId: 'bfd-aws-account-id', variable: 'ACCOUNT_ID')]) {
		// Set the auth token for aws code artifact
		env.CODEARTIFACT_AUTH_TOKEN = sh(
			returnStdout: true,
			script: '''
aws codeartifact get-authorization-token --domain bfd-mgmt \
 --domain-owner "$ACCOUNT_ID" \
 --output text --query authorizationToken
'''
		).trim()

		// Get our endpoint url for our aws code artifact
		env.CODEARTIFACT_ENDPOINT = sh(
			returnStdout: true,
			script: '''
aws codeartifact get-repository-endpoint \
--domain bfd-mgmt --repository bfd-mgmt \
--format maven --output text
'''
		).trim()
	}

	// Add the authorization token and username for our aws code artifact repository
	sh '''
cat <<EOF > ~/.m2/settings.xml
<settings xmlns=\"http://maven.apache.org/SETTINGS/1.0.0\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"
xsi:schemaLocation=\"http://maven.apache.org/SETTINGS/1.0.0 https://maven.apache.org/xsd/settings-1.0.0.xsd\">
    <profiles>
        <profile>
            <id>bfd-mgmt-bfd-mgmt</id>
            <activation>
            <activeByDefault>true</activeByDefault>
            </activation>
            <repositories>
                <repository>
                    <id>bfd-mgmt-bfd-mgmt</id>
                    <url>${CODEARTIFACT_ENDPOINT}</url>
                </repository>
            </repositories>
        </profile>
    </profiles>
    <servers>
        <server>
        <username>aws</username>
        <password>${CODEARTIFACT_AUTH_TOKEN}</password>
        <id>bfd-mgmt-bfd-mgmt</id>
        </server>
    </servers>
</settings>
EOF
'''

	dir ('apps') {
		quietFlags = verboseMaven ? '' : '--quiet --batch-mode'
		sh "mvn ${quietFlags} --threads 1C --update-snapshots -DskipITs -DskipTests -Dmaven.javadoc.skip=true clean verify"
	}
```